python/tank/authentication/user_impl.py

```diff
@@ -361,7 +361,7 @@ def __str__(self):
 
         :returns: A string.
         """
-        return self._login
+        return six.ensure_str(self._login)
```

Comment:
I don't think we need try-except at this point because it only applies when the login is previously serialized like in the dumps method of the pickle custom module.